cmd/optimize/events.go

```diff
@@ -337,33 +336,38 @@ func followDatasetAndPrint(cmd *cobra.Command, data_set *uql.DataSet, no_results
 	main_data_set := resp.Main()
 	if main_data_set == nil {
 		log.Error("Following of events query has nil main data. Returned data may not be complete!")
-		return data_set, nil
+		return &followEventResult{data_set: data_set}
 	}
 	if len(main_data_set.Data) < 1 {
-		return nil, fmt.Errorf("follow main dataset %v has no rows", main_data_set.Name)
+		return &followEventResult{err: fmt.Errorf("follow main dataset %v has no rows", main_data_set.Name)}
 	}
 	if len(main_data_set.Data[0]) < 1 {
-		return nil, fmt.Errorf("follow main dataset %v first row has no columns", main_data_set.Name)
+		return &followEventResult{err: fmt.Errorf("follow main dataset %v first row has no columns", main_data_set.Name)}
 	}
 	var ok bool
 	data_set, ok = main_data_set.Data[0][0].(*uql.DataSet)
 	if !ok {
-		return nil, fmt.Errorf("follow main dataset %v first row first column (type %T) could not be converted to *uql.DataSet", main_data_set.Name, main_data_set.Data[0][0])
+		return &followEventResult{err: fmt.Errorf("follow main dataset %v first row first column (type %T) could not be converted to *uql.DataSet", main_data_set.Name, main_data_set.Data[0][0])}
 	}
 
+	result := &followEventResult{data_set: data_set}
 	newRows, err := extractEventsData(data_set)
 	if err != nil {
-		return nil, fmt.Errorf("follow extractEventsData: %w", err)
+		result.err = fmt.Errorf("follow extractEventsData: %w", err)
+		return result
 	}
-	if newRowsCount := len(newRows); newRowsCount > 0 {
+
+	newRowsCount := len(newRows)
+	if newRowsCount > 0 {
 		output.PrintCmdOutputAdvanced(cmd, struct {
 			Items []eventsRow `json:"items"`
 			Total int         `json:"total"`
 		}{Items: newRows, Total: newRowsCount}, &output.PrintRequest{HideTableHeaders: true})
-	} else {
-		*no_results_flag = true
 	}
-	return data_set, nil
+	if newRowsCount < 1000 {
```

Comment:
One last thing. This code can still diverge from the intended behavior at any time. There is no specification for what is the default returned number of rows from a UQL query. More importantly, a fsoc user can change the limit with the flag `--count`. Please, don't use the literal 1000 in the condition

Reply:
Done